gradio/blocks.py

```diff
@@ -361,6 +363,7 @@ def __init__(
         self.auth = None
         self.dev_mode = True
         self.app_id = random.getrandbits(64)
+        self.temp_dir = tempfile.TemporaryDirectory().name
```

Comment:
One big change is that we now don't serialize output files (audio, videos, etc.) to base64 format to send them to the frontend. Instead, we save the files to a temporary directory and simply return the file link (linking is faster than serialization and this approach consolidates code). The `self.temp_dir` the location for these temporary files for a given Blocks.